### DISCUSSION TITLE
Rhino 1.7.14 RC1

### DISCUSSION THREAD

**Discussion by gbrail** [mozilla/rhino#1088] [Announcements]
I just published  [Rhino 1.7.14 RC1](https://github.com/mozilla/rhino/releases/tag/Rhino1_7_14_RC1_Release) on GitHub and on Maven Central (give it a few hours if you don't see it there yet).

Thanks to so many of you who contributed!

Next steps:

* Please use the next two weeks to try out this release with your workloads and open issues or PRs if it introduces new bugs.
* I will not be merging PRs into master unless they fix a regression or another major bug.
* In about two weeks, if there are no issues, I'll publish again as 1.7.14, and then the community can start to discuss what to do next.

**Comment by p-bakker** [comment]
I know it's quite late to bring this up, but #995 is a big related to BigInt, which got added in 1.7.14.

Should we not triage it and if considered a bug, fix it before the final release of 1.7.14, as to not have to do a potential breaking fix for it in a later release?

**Reply by p-bakker** [reply]
@gbrail thoughts?

**Reply by p-bakker** [reply]
@tonygermano @tuchida 

Thoughts on fixing this before our BigInt is released out into the wild?

Is there any discussion on what the correct behavior ought to be?

Is the fix straight forward or tricky, in the sense that it'll only affect the codepath in case it's dealing with NativeBigInt?

Basically: what's the risk of fixing this?

**Reply by tuchida** [reply]
Can I fix it now?
`java.math.BigInteger` as `BigInt`, so there is a compatibility problem that the handling of `java.lang.Number` has changed(ref. #837 https://github.com/mozilla/rhino/pull/837#issuecomment-818307359). Fixing #995 will result in a larger breaking change.

**Reply by p-bakker** [reply]
> so there is a compatibility problem that the handling of java.lang.Number has changed(ref. #837 #837 (comment)

Not exactly sure I get what you're saying, sry. Are you saying that in v1.7.14 there's a behavior change in Rhino that we should document in the release notes?

 > Can I fix it now? .... Fixing #995 will result in a larger breaking change.
 
I think if we deem that #995 needs to be fixed eventually, we'd be better of doing it now, because this release is where BigInt was introduced.
But I'd like others to chime in on that

**Reply by tuchida** [reply]
I'll try to fix it for now. Please look at the pull request to decide whether to merge or not.

**Comment by p-bakker** [comment]
@botic in #966 you mentioned your give the RC a spin in RingoJS. Any update on that?

**Reply by botic** [reply]
[Updated our master branch](https://github.com/ringo/ringojs/commit/231ab637e21ca50b0087d0789a0fade10d6a915d) and so far no problems. I'll test in the next three days in the office some more exotic applications with it.

**Reply by p-bakker** [reply]
HtmlUnit also reporting positive on the Rhino Promise implementation: https://github.com/mozilla/rhino/pull/942#issuecomment-983337909

**Reply by botic** [reply]
We see some minor troubles with **our** old RingoJS Promise "workaround" implementation, but this is nothing you should care about.

**Reply by botic** [reply]
Until now no bigger issues. I run a larger application on 1.7.14 and no problems so far. We might see some more stuff if we release Ringo's own 3.0.0 since a lot of production code actually is on Ringo's 1.2 release.

**Comment by p-bakker** [comment]
A request to include a PR in 1.7.14: https://github.com/mozilla/rhino/issues/1100#issuecomment-981065556

Can't judge myself whether we should or not

**Reply by tonygermano** [reply]
This does not seem urgent, however, it doesn't actually change any code, so I don't see much harm in it, either.

If it's only a warning during compilation, I would expect anyone compiling from source to be compiling from the current master branch and not the tag of the most recent release.

**Reply by p-bakker** [reply]
This got merged

**Comment by p-bakker** [comment]
@gbrail did you intentionally not include the rhino-runtime/engine jars in the RC1 release?

**Comment by p-bakker** [comment]
One regression left to tackle: #1111

And potentially #995, see https://github.com/mozilla/rhino/discussions/1088#discussioncomment-1711243
